src/tlo/methods/wasting.py

```diff
@@ -1,41 +1,1374 @@
+            else:
+                # remained MAM
+                return
+
+        elif intervention in ['OTC', 'ITC']:
+            if intervention == 'OTC':
+                outcome_date = (df.at[person_id, 'un_am_tx_start_date'] +
+                                DateOffset(weeks=p['tx_length_weeks_OutpatientSAM']))
+            else:
+                outcome_date = (df.at[person_id, 'un_am_tx_start_date'] +
+                                DateOffset(weeks=p['tx_length_weeks_InpatientSAM']))
+
+            sam_full_recovery = self.wasting_models.acute_malnutrition_recovery_sam_lm.predict(
+                df.loc[[person_id]], self.rng
+            )
+            if sam_full_recovery:
+                # schedule full recovery
+                self.sim.schedule_event(
+                    event=Wasting_ClinicalAcuteMalnutritionRecovery_Event(module=self, person_id=person_id),
+                    date=outcome_date
+                )
+                # cancel death date
+                df.at[person_id, 'un_sam_death_date'] = pd.NaT
+            else:
+                outcome = self.rng.choice(['recovery_to_mam', 'death'], p=[self.parameters['prob_mam_after_care'],
+                                                                           self.parameters['prob_death_after_care']])
+                if outcome == 'death':
+                    self.sim.schedule_event(
+                        event=Wasting_SevereAcuteMalnutritionDeath_Event(module=self, person_id=person_id),
+                        date=outcome_date
+                    )
+                else:  # recovery to MAM and follow-up treatment for MAM
+                    self.sim.schedule_event(event=Wasting_UpdateToMAM_Event(module=self, person_id=person_id),
+                                            date=outcome_date)
+                    self.sim.modules['HealthSystem'].schedule_hsi_event(
+                        hsi_event=HSI_Wasting_SupplementaryFeedingProgramme_MAM(module=self, person_id=person_id),
+                        priority=0, topen=outcome_date)
+                    # cancel death date
+                    df.at[person_id, 'un_sam_death_date'] = pd.NaT
+
+
+class Wasting_IncidencePoll(RegularEvent, PopulationScopeEventMixin):
+    """
+    Regular event that determines new cases of wasting (WHZ < -2) to the under-5 population, and schedules
+    individual incident cases to represent onset. It determines those who will progress to severe wasting
+    (WHZ < -3) and schedules the event to update on properties. These are events occurring without the input
+    of interventions, these events reflect the natural history only.
+    """
+    AGE_GROUPS = {0: '0y', 1: '1y', 2: '2y', 3: '3y', 4: '4y'}
+
+    def __init__(self, module):
+        """schedule to run every month
+        :param module: the module that created this event
+        """
+        self.repeat_months = 1
+        super().__init__(module, frequency=DateOffset(months=self.repeat_months))
```

Comment:
@thewati, iIs there any reason to have the `repeat_months` as a parameter instead of setting here the frequency just like:
`frequency=DateOffset(months=1)`?